tests/components/airzone/test_config_flow.py

```diff
@@ -145,3 +157,194 @@ async def test_connection_error(hass: HomeAssistant):
+        return_value=HVAC_MOCK,
+    ), patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_hvac_systems",
+        side_effect=SystemOutOfRange,
+    ), patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_webserver",
+        return_value=HVAC_WEBSERVER_MOCK,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PORT: TEST_PORT,
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state is ConfigEntryState.LOADED
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == f"Airzone {short_mac(HVAC_WEBSERVER_MOCK[API_MAC])}"
+        assert result["data"][CONF_HOST] == TEST_IP
+        assert result["data"][CONF_PORT] == TEST_PORT
+
+        mock_setup_entry.assert_called_once()
+
+
+async def test_dhcp_invalid_system_id(hass: HomeAssistant) -> None:
+    """Test Invalid System ID 0."""
+
+    with patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_version",
+        return_value=HVAC_VERSION_MOCK,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DHCP_SERVICE_INFO,
+            context={"source": config_entries.SOURCE_DHCP},
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "discovered_connection"
+
+    with patch(
+        "homeassistant.components.airzone.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_hvac",
+        side_effect=InvalidSystem,
+    ) as mock_hvac, patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_hvac_systems",
+        side_effect=SystemOutOfRange,
+    ), patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_webserver",
+        side_effect=InvalidMethod,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PORT: TEST_PORT,
+            },
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "discovered_connection"
+        assert result["errors"] == {CONF_ID: "invalid_system_id"}
+
+        mock_hvac.return_value = HVAC_MOCK[API_SYSTEMS][0]
+        mock_hvac.side_effect = None
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PORT: TEST_PORT,
+                CONF_ID: TEST_ID,
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state is ConfigEntryState.LOADED
```

Comment:
Since you are patching setup this check is likely redundant but it's not hurting anything